pandas/core/indexing.py

```diff
@@ -260,11 +267,11 @@ def _convert_tuple(self, key, is_setter: bool = False):
                 keyidx.append(idx)
         return tuple(keyidx)
 
-    def _convert_range(self, key, is_setter: bool = False):
+    def _convert_range(self, key: range, is_setter: bool = False) -> List[int]:
         """ convert a range argument """
         return list(key)
 
-    def _convert_scalar_indexer(self, key, axis: int):
+    def _convert_scalar_indexer(self, key: Any, axis: int) -> Any:
```

Comment:
this isn't restricted to scalar key?

Reply:
` monkeytype --disable-type-rewriting` gives

```
    def _convert_scalar_indexer(
        self,
        key: Optional[Union[float64, List[str], str, ndarray, bytes, RangeIndex, List[Tuple[str, str]], Tuple[slice, slice], Dict[Tuple[str, str], int], Timedelta, bool, Tuple[str, int], Float64Index, int, Tuple[slice, slice, str], TimedeltaIndex, Tuple[int64, int64, int64, int64], Dict[str, int], Int64Index, int32, List[int], Index, float, List[Any], time, Tuple[int, int], Series, Set[str], int64, Interval, Tuple[slice, slice, List[str]], Tuple[slice, int], DatetimeIndex, SparseArray, Tuple[slice, List[int]], Timestamp, List[Timestamp], datetime64, Set[Tuple[str, str]], MultiIndex, datetime, Tuple[str, str], Tuple[slice, List[str]], NaTType, List[bool]]],
        axis: int
    ) -> Optional[Union[float64, List[str], str, ndarray, bytes, RangeIndex, List[Tuple[str, str]], Tuple[slice, slice], Dict[Tuple[str, str], int], Timedelta, bool, Tuple[str, int], Float64Index, int, Tuple[slice, slice, str], TimedeltaIndex, Tuple[int64, int64, int64, int64], Dict[str, int], Int64Index, int32, List[int], Index, float, List[Any], time, Tuple[int, int], Series, Set[str], int64, Interval, Tuple[slice, slice, List[str]], Tuple[slice, int], DatetimeIndex, SparseArray, Tuple[slice, List[int]], Timestamp, List[Timestamp], datetime64, Set[Tuple[str, str]], MultiIndex, datetime, Tuple[str, str], Tuple[slice, List[str]], NaTType, List[bool]]]: ...
```

I could be wrong, but I assume that test functions that raise are not included in the traces.  There wouldn't much point otherwise.

Reply:
So no way to restrict this further? `Any` just doesn't really provide the reader/developer with any more insight into how the function is supposed to work.

Reply:
> So no way to restrict this further?

probably. but not time well spent until more call sites and more imports are typed.